src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -745,29 +719,29 @@ seastar::future<> ReplicatedRecoveryBackend::_handle_push(
 {
   logger().debug("{}", __func__);
 
-  return seastar::do_with(interval_set<uint64_t>(),
-			  bufferlist(),
-    [this, &pop, t, response](auto& data_zeros, auto& data) {
-    data = pop.data;
-    bool first = pop.before_progress.first;
-    bool complete = pop.after_progress.data_complete
-      && pop.after_progress.omap_complete;
-    bool clear_omap = !pop.before_progress.omap_complete;
-    uint64_t z_offset = pop.before_progress.data_recovered_to;
-    uint64_t z_length = pop.after_progress.data_recovered_to
-      - pop.before_progress.data_recovered_to;
-    if (z_length)
-      data_zeros.insert(z_offset, z_length);
-    response->soid = pop.recovery_info.soid;
+  bool first = pop.before_progress.first;
+  interval_set<uint64_t> data_zeros;
+  {
+    uint64_t offset = pop.before_progress.data_recovered_to;
+    uint64_t length = (pop.after_progress.data_recovered_to -
+                       pop.before_progress.data_recovered_to);
+    if (length) {
+      data_zeros.insert(offset, length);
+    }
+  }
+  bool complete = (pop.after_progress.data_complete &&
+		   pop.after_progress.omap_complete);
+  bool clear_omap = !pop.before_progress.omap_complete;
+  response->soid = pop.recovery_info.soid;
 
-    return submit_push_data(pop.recovery_info, first, complete, clear_omap,
-	data_zeros, pop.data_included, data, pop.omap_header, pop.attrset,
-	pop.omap_entries, t).then([this, complete, &pop, t] {
-      if (complete) {
-	pg.get_recovery_handler()->on_local_recover(pop.recovery_info.soid,
-			    pop.recovery_info, false, *t);
-      }
-    });
+  return submit_push_data(pop.recovery_info, first, complete, clear_omap,
+        data_zeros, pop.data_included, pop.data, pop.omap_header,
```

Comment:
data_zeros may be captured by reference in [submit_push_data](https://github.com/ceph/ceph/blob/7ef6453e4aff9bcef8bb2dd26182dd35bb98dee0/src/crimson/osd/replicated_recovery_backend.cc#L989), I think that may also need to be modified

Reply:
@xxhdx1985126 fixed and repushed.